lib/bitshares.js

```diff
@@ -198,6 +198,43 @@ export default class BitShares {
     }
   }
 
+    async memoDecode(memos) {
+        if (!this.memoKey)
+            throw new Error("Not set memoKey!");
+
+        let memo_text;
+        let pub_key = await PublicKey.fromPublicKeyString(memos.from);
+        let prv_key = this.memoKey;
+        // console.log("\nPrivate key:", prv_key.toWif());
+        console.log("Public key :", prv_key.toPublicKey().toString(), "\n");
+
+        if (prv_key) {
+            try {
+                memo_text = prv_key
+                    ? await Aes.decrypt_with_checksum(
+                        prv_key,
+                        pub_key,
+                        memos.nonce,
+                        memos.message
+                    ).toString("utf-8")
+                    : null;
+
+                if (prv_key && !memo_text) {
+                    // debugger
+                }
+            } catch (e) {
+                console.log("transfer memo exception ...", e);
+                memo_text = false;
+            }
+
+        }
+
+
+        return {
+            message: memo_text
+        }
```

Comment:
Why return an object?

Reply:
oh yes, you are right, it is better to return the text and remove the superfluous condition